pycape/function_ref.py

```diff
@@ -109,3 +109,21 @@ def from_json(cls, token_path: Union[str, os.PathLike]):
         function_checksum = token_config.get("function_checksum")
 
         return cls(function_id, function_token, function_checksum)
+
+    def to_json(self, path: Optional[os.PathLike]):
```

Comment:
this type signature looks incorrect for the code that's here, however I think I like this pattern better. If `path=None`, return the json as a string, otherwise if path is convertible to a filepath write the json to that location (as it is here)

Reply:
we could then add the same pattern to `from_json`, so that round-tripping these to/from json is possible w/out saving to a file